src/main/java/com/artipie/metrics/MetricsContext.java

```diff
@@ -0,0 +1,300 @@
+        final YamlSequence seq = settings.yamlSequence(MetricsContext.METRICS);
+        if (seq != null) {
+            seq.values().stream()
+                .map(YamlNode::asMapping)
+                .forEach(
+                    mapping -> {
+                        final String name = mapping.string("type");
+                        if (!Strings.isNullOrEmpty(name)) {
+                            final MetricsOutputType type = MetricsOutputType.valueOf(
+                                name.toUpperCase(Locale.getDefault())
+                            );
+                            if (!this.outputs.containsKey(type)) {
+                                this.outputs.put(type, mapping);
+                            } else {
+                                Logger.error(
+                                    MetricsContext.class,
+                                    "Metric with type %s already exists",
+                                    name
+                                );
+                            }
+                        } else {
+                            throw new IllegalArgumentException("Empty metric type is not allowed");
+                        }
+                    }
+            );
+        }
+        Logger.info(MetricsContext.class, "Metrics [outputs=%s]", this.outputs);
+        if (!this.outputs.isEmpty()) {
+            this.metrics = new InMemoryMetrics();
+            final Collection<IntervalMetricOutput> list = this.outputs.entrySet()
+                .stream()
+                .filter(
+                    entry -> entry.getKey().isInterval()
+                )
+                .map(
+                    entry -> {
+                        final IntervalMetricOutput res;
+                        switch (entry.getKey()) {
+                            case LOG:
+                                res = new MetricsLogOutput(
+                                    LoggerFactory.getLogger(Metrics.class),
+                                    interval(entry.getValue())
+                                );
+                                break;
+                            case ASTO:
+                                res = new StorageMetricsOutput(
+                                    metricsStorage(entry.getValue()
+                                        .yamlMapping(MetricsContext.STORAGE)
+                                    ),
+                                    interval(entry.getValue())
+                                );
+                                break;
+                            default:
+                                throw new IllegalStateException(
+                                    String.format(
+                                        "%s is not interval type of metrics output",
+                                        entry.getValue()
+                                    )
+                                );
+                        }
+                        return res;
```

Comment:
I'd suggest simplifying this part by using enhanced switch (follow Idea's suggestions, it will bring you there:))

Reply:
Unfortunately, `Qulice` doesn't know about enhanced switch and falls down.